#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "social_compliance_layer"

gen = ParameterGenerator()

gen.add('enabled', bool_t, 0, 'Whether to apply this plug-in or not', True)
gen.add('scaled', bool_t, 0, 'Scaling influence regions', False)
gen.add('dynamic_scene', bool_t, 0, 'Switch for dynamic scenarios', True)
gen.add('region_type', str_t, 0,
        'Influence region type (uniform or anisotropic)', 'anisotropic')
gen.add('behavior_name', str_t, 0,
        'Name of behavior (polite, sociable or rude)', 'polite')
gen.add('thresh_link', double_t, 0,
        'Max relation influence reach', 1.2, 0.0, 10.0)
gen.add('thresh_person', double_t, 0,
        'Max personal influence reach', 1.2, 0.0, 10.0)
gen.add('thresh_static', double_t, 0,
        'Velocity threshold for moving vs static', 0.4, 0.1, 3.0)
gen.add('human_radius', double_t, 0,
        'Radius of a person', 0.24, 0.1, 0.5)
gen.add('update_range', double_t, 0,
        'How far from the robot to update', 8.0, 1.0, 100.0)
gen.add('min_range', double_t, 0,
        'How near to the robot to update', 0.3, 1.0, 10.0)
gen.add('prediction_horizon', int_t, 0,
        'Prediction horizon into the future', 2, 0, 10)
exit(gen.generate(PACKAGE, PACKAGE, "SocialComplianceLayer"))
